src/applicationcomposer/messageHandlers/deployMessageHandler.ts

```diff
@@ -32,19 +33,18 @@ async function promptReauth(connection?: StatefulConnection) {
         'Authenticate with IAM credentials'
     )
     const docMessage = localize('aws.applicationComposer.deploy.authModal.docLink', 'Open documentation')
-    const modalResponse = await vscode.window.showInformationMessage(
+    const modalResponse = await showMessageWithUrl(
         errorMessage,
-        { modal: true },
-        acceptMessage,
-        docMessage
+        credentialHelpUrl,
+        docMessage,
+        'info',
+        [acceptMessage],
+        true
```

Comment:
We want a modal here because the user has decided that they want to sync their IaC, and we need user action before we can proceed. The sync will not work until the user switches auth. The toast notification feels too easy to miss, as it is in the corner with other notifications and automatically disappears after some time.

From a code logic perspective, the toast notification would also work, but the `await` does not seem to complete execution when the notification disappears from timeout.

Reply:
> The sync will not work until the user switches auth. The toast notification feels too easy to miss, as it is in the corner with other notifications and automatically disappears after some time.

"Too easy to miss" is a feature. Modal dialogs are very rare in vscode, intentionally. The toast behavior is configurable. 

It doesn't sound like this needs to be modal, because the action will be canceled correctly, and this is only informational.

Reply:
The way it's currently implemented, it is not only informational. The user is marked as being "in progress of syncing template" until they explicitly cancel, and the re-authentication is considered as a step in the sync process. That's not to say we can't use a toast notification instead, but I think the flow would need to be a bit different to be intuitive.

Current flow for reauthenticating with modal:
Sync -> See modal -> Re-auth -> standard SAM Sync prompter

Given the passive nature of the toast notifications, I don't think it makes sense to include as part of the flow. Instead, I could see something like this being an option:
Sync -> See toast -> Re-auth -> Sync again -> standard SAM Sync prompter

Would the latter approach be preferable?

Reply:
What about this:

    Sync -> See toast (informational only) -> Prompt to select credentials -> standard SAM Sync prompter